server/server_test.go

```diff
@@ -366,6 +366,104 @@ func (s *ServerSuite) TestUpdatePatient(c *C) {
 	c.Assert(since.Minutes() < float64(1), Equals, true)
 }
 
+func (s *ServerSuite) TestConditionalUpdatePatientNoMatch(c *C) {
+	data, err := os.Open("../fixtures/patient-example-c.json")
+	util.CheckErr(err)
+	defer data.Close()
+
+	req, err := http.NewRequest("PUT", s.Server.URL+"/Patient?name=Donny", data)
+	req.Header.Add("Content-Type", "application/json")
+	util.CheckErr(err)
+	res, err := http.DefaultClient.Do(req)
+
+	c.Assert(res.StatusCode, Equals, 201)
+	splitLocation := strings.Split(res.Header["Location"][0], "/")
+	createdPatientID := splitLocation[len(splitLocation)-1]
+
+	patientCollection := s.Database.C("patients")
+	count, err := patientCollection.Count()
+	util.CheckErr(err)
+	c.Assert(count, Equals, 2)
+
+	// Check new patient
+	patient := models.Patient{}
+	err = patientCollection.FindId(createdPatientID).One(&patient)
+	util.CheckErr(err)
+	c.Assert(patient.Name[0].Given[0], Equals, "Donny")
+	c.Assert(patient.Meta, NotNil)
+	c.Assert(patient.Meta.LastUpdated, NotNil)
+	c.Assert(patient.Meta.LastUpdated.Precision, Equals, models.Precision(models.Timestamp))
+	since := time.Since(patient.Meta.LastUpdated.Time)
+	c.Assert(since.Hours() < float64(1), Equals, true)
+	c.Assert(since.Minutes() < float64(1), Equals, true)
+
+	// Check existing (unmatched) patient
+	patient2 := models.Patient{}
+	err = patientCollection.FindId(s.FixtureID).One(&patient2)
+	util.CheckErr(err)
+	c.Assert(patient2.Name[0].Given[0], Equals, "Donald")
+}
+
+func (s *ServerSuite) TestConditionalUpdatePatientOneMatch(c *C) {
+	data, err := os.Open("../fixtures/patient-example-c.json")
+	util.CheckErr(err)
+	defer data.Close()
+
+	req, err := http.NewRequest("PUT", s.Server.URL+"/Patient?name=Donald", data)
+	req.Header.Add("Content-Type", "application/json")
+	util.CheckErr(err)
+	res, err := http.DefaultClient.Do(req)
+
+	c.Assert(res.StatusCode, Equals, 200)
+	patientCollection := s.Database.C("patients")
+	count, err := patientCollection.Count()
+	util.CheckErr(err)
+	c.Assert(count, Equals, 1)
+	patient := models.Patient{}
+	err = patientCollection.FindId(s.FixtureID).One(&patient)
+	util.CheckErr(err)
+	c.Assert(patient.Name[0].Given[0], Equals, "Donny")
+	c.Assert(patient.Meta, NotNil)
+	c.Assert(patient.Meta.LastUpdated, NotNil)
+	c.Assert(patient.Meta.LastUpdated.Precision, Equals, models.Precision(models.Timestamp))
+	since := time.Since(patient.Meta.LastUpdated.Time)
+	c.Assert(since.Hours() < float64(1), Equals, true)
+	c.Assert(since.Minutes() < float64(1), Equals, true)
```

Comment:
Um.  Yeah.  Not sure what I was thinking the first time I wrote that.  And then I copied/pasted it a few times. =/

I can remove it -- but I have to leave early today, so will have to be tomorrow.